### PR TITLE
MethodGenerator::fromArray() add config keys

### PR DESCRIPTION
**Pull request by WinterSilence**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Add keys `'returntype'/setReturnType()` and  `'returnsreference'/setReturnsReference()` to array passed in `MethodGenerator::fromArray()`.

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
Hey @WinterSilence, could you please add a test for the `'returnsreference'` scenario?

**Comment by WinterSilence** [comment]
@Ocramius ok, tomorrow.

**Comment by WinterSilence** [comment]
@Ocramius fixed but something wrong in code style, can you help?

File: src/Generator/MethodGenerator.php [hunk 1]
```diff
@@ -170,6 +172,8 @@ public static function fromArray(array $array)
                 case 'returntype':
                     $method->setReturnType($value);
                     break;
+                case 'returnsreference':
+                    $method->setReturnsReference($value);
```
Comment (Ocramius) [inline src/Generator/MethodGenerator.php:176]: I think we need a `(bool)` cast here.

Alternatively, we can declare `@param array<...>`  with more specific types, but that will likely open another rabbit hole

File: test/Generator/MethodGeneratorTest.php [hunk 2]
```diff
@@ -283,7 +283,7 @@ public function testDefaultValueGenerationDoesNotIncludeTrailingSemicolon()
 
     public function testCreateFromArray()
     {
-        $methodGenerator = MethodGenerator::fromArray([
+        $config = [
```
Comment (Ocramius) [inline test/Generator/MethodGeneratorTest.php:286]: Run a `vendor/bin/phpcbf` and it should fix any issues around this :)

File: test/Generator/MethodGeneratorTest.php [hunk 3]
```diff
@@ -305,6 +306,11 @@ public function testCreateFromArray()
         self::assertSame(MethodGenerator::VISIBILITY_PROTECTED, $methodGenerator->getVisibility());
         self::assertInstanceOf(TypeGenerator::class, $methodGenerator->getReturnType());
         self::assertSame('\\SampleType', $methodGenerator->getReturnType()->generate());
+        self::assertFalse($methodGenerator->isReturnsReference());
+
+        $config['returnsreference'] = true;
```
Comment (Ocramius) [inline test/Generator/MethodGeneratorTest.php:311]: IMO, we should have a separate (new) test method to verify this one

File: test/Generator/MethodGeneratorTest.php [hunk 4]
```diff
@@ -305,6 +306,11 @@ public function testCreateFromArray()
...
+        $methodGenerator = MethodGenerator::fromArray($config);
```
Comment (Ocramius) [inline test/Generator/MethodGeneratorTest.php:312]: Also here, `vendor/bin/phpcbf` will fix most issues

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Brammm** [comment]
Is there something I can help with to get this PR through, @WinterSilence?

File: src/Generator/MethodGenerator.php (hunk 1 above)
Comment (Brammm) [reply src/Generator/MethodGenerator.php:176]: How about the existing methods, like setBody, setVisibility (string) or setAbstract, setFinal, setInterface, setStatic (bools)? Should those get relevant type casts as well?

**Comment by Brammm** [comment]
I've addressed @Ocramius' comments in this branch (I think): https://github.com/Brammm-forks/laminas-code/tree/patch-1 I can open a new PR or they can be merged into this one, I guess. Whatever is fine with me.

**Comment by WinterSilence** [comment]
@Brammm PR sended 2 months ago... now, I don't care anymore

**Comment by Ocramius** [comment]
Merged via #121

**Pull request closed without merging**